builtin/mainmenu/dlg_create_world.lua

```diff
@@ -102,20 +102,31 @@ local function create_world_button_handler(this, fields)
 		map_generator = fields["map_generator"]
 	}
 
-	if buttons.create then
-		if values.name == "" or values.game_index == nil then
-			gamedata.errormessage =
-				fgettext("No worldname given or no game selected")
-			return true
+	if values.name == "" then
+		values.name = fgettext_ne("Unnamed")
+	end
+	while menudata.worldlist:uid_exists_raw(values.name) do
+		local words = {}
+		for word in string.gmatch(values.name, "[^ ]+") do
+			table.insert(words, word)
+		end
+		local number = tonumber(words[#words])
+		if number == nil then
+			values.name = values.name .. " 2"
```

Comment:
Not keen on a space, as the space will appear in the world filename and spaces in filenames are not good.